### PR TITLE
Refactor and Replace New Statement Upload States

### PR DESCRIPTION
**Pull request by li-xinyang**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,26 +1,21 @@
 {
   "name": "money-cat",
-  "version": "0.1.0",
+  "version": "1.0.0",
```
Comment (weilu) [inline package.json]: 🚀woot! 🚀

File: src/states/actions/statements.js [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 export const ACTION_TYPE = {
   UPLOAD_STATEMENT_SUCCESSFUL: "UPLOAD_STATEMENT_SUCCESSFUL",
   UPLOAD_STATEMENT_FAILED: "UPLOAD_STATEMENT_FAILED",
 
-  DELETE_STATEMENT_TRANSACTION: "DELETE_STATEMENT_TRANSACTION"
+  DELETE_STATEMENT_TRANSACTION: "DELETE_STATEMENT_TRANSACTION",
+  UPDATE_STATEMENT_TRANSACTION: "UPDATE_STATEMENT_TRANSACTION",
+
+  UPLOAD_STATEMENT_TRANSACTIONS_SUCCESSFUL: "UPLOAD_STATEMENT_TRANSACTIONS_SUCCESSFUL",
+  UPLOAD_STATEMENT_TRANSACTIONS_FAILED: "UPLOAD_STATEMENT_TRANSACTIONS_FAILED"
```
Comment (weilu) [inline src/states/actions/statements.js]: So I guess the difference between UPLOAD_STATEMENT_SUCCESSFUL and UPLOAD_STATEMENT_TRANSACTIONS_SUCCESSFUL is upload pdf for parsing (doesn't change server side state) vs. actually sending transactions for server to save? Maybe give them better names to differentiate them, such as PARSE_STATEMENT vs. ADD/UPLOAD_STRANSACTIONS?

File: src/states/actions/statements.js [hunk 3]
```diff
@@ -1,10 +1,14 @@
-import { requestUploadStatementFile } from "../../utilities/api";
+import { requestUploadStatementFile, requestUploadTransactions } from "../../utilities/api";
 
 export const ACTION_TYPE = {
   UPLOAD_STATEMENT_SUCCESSFUL: "UPLOAD_STATEMENT_SUCCESSFUL",
   UPLOAD_STATEMENT_FAILED: "UPLOAD_STATEMENT_FAILED",
 
-  DELETE_STATEMENT_TRANSACTION: "DELETE_STATEMENT_TRANSACTION"
+  DELETE_STATEMENT_TRANSACTION: "DELETE_STATEMENT_TRANSACTION",
+  UPDATE_STATEMENT_TRANSACTION: "UPDATE_STATEMENT_TRANSACTION",
```
Comment (weilu) [inline src/states/actions/statements.js]: Doesn't update have failure state?

File: src/states/actions/statements.js (hunk 2 above)
Comment (li-xinyang) [reply src/states/actions/statements.js]: WIP

**Pull request merged**